articles/embedded.md

```diff
@@ -0,0 +1,309 @@
+---
+layout: default
+title: Embedded ROS 2.0
+abstract:
+  This article captures requirements, design ideas and related works on a tiny ROS 2.0 stack for microcontrollers.
+published: true
+author: 'In alphabetic order: [Adam Dąbrowski](https://github.com/adamdbrw), [Borja Outerelo](https://github.com/BorjaOuterelo), [Ingo Lütkebohle](https://github.com/iluetkeb), [Ralph Lange](https://github.com/ralph-lange), [Víctor Mayoral Vilches](https://github.com/vmayoral), ... and many more from Embedded ROS 2 Interest Group. Please feel free to add your name.'
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+While standard ROS 2 can run on Linux-class embedded devices, smaller 
+microcontrollers (MCU) are still a challenge. For example, ARM Cortex-M3 
+and -M4 class MCUs are popular for motion control and sensor integration, 
+but have little RAM and storage. ARM Cortex-M3 is widely used in many 
+sensors due to its optimized power consumption capabilities and ultra low 
+power modes. These microcontrollers are often running with small batteries, 
+e.g. for environmental sensors or small consumer robots, which requires
+efficient power management. We aim to address these devices through a 
+combination of specialized, ROS 2-interoperable stack as well as the use
+of Real-Time Operating Systems (RTOS).
+
+This article documents feature wishes, requirements, design ideas and 
+related work on this undertaking.
+
+## Wishlist
+
+A number of things that would be great to have, lets call it a wishlist. 
+Feel free to add stuff ;-)
+
+### W-SEAMLESS: *Seamless integration with ROS&nbsp;2*
+
+The embedded ROS 2 stack shall integrate seamlessly with standard 
+ROS 2 stacks and nodes running on more powerful microprocessors. For 
+a standard ROS 2 node, communication (topics, services) from and to
+software running on the embedded ROS 2 stack should be transparent. 
+The same should hold for other core concepts such as parameters, 
+graph introspection, and run-time reconfiguration by the node lifecycle.
+
+### W-PORTABILITY: *Portability of ROS 2-based software*
+
+The embedded ROS 2 stack shall resemble or directly use the ROS 2 API 
+-- more precisely the rclcpp API -- to facilitate porting standard 
```

Comment:
This implies you want to have a C++ API but find the current rclcpp implementation too heavy for tiny scale devices. Is that why you are creating a separate implementation? If so, why not improve rmw/rcl and then create a separate C++ client library built on top of them, e.g. rclcpp_tiny? This would preserve the benefits of everyone wrapping rcl, bring the improved performance benefits of this project to all client libraries, and still satisfy your desire for a C++ API implementation oriented to low-resource systems.

Reply:
This implies merely that we haven't made up our mind, yet, on what it's going to be.